### PR TITLE
Reformat based on clang-format's LLVM style with 4 spaces

### PR DESCRIPTION
**Pull request by timwoj**
I'm working on updating the version of the patricia code we use in https://github.com/zeek/zeek. I noticed that the formatting in the code is all over the place. It's a mix of tabs and spaces and varying indentation styles. This PR reformats it using clang-format and `--style="{BasedOnStyle: llvm, IndentWidth: 4}"`. There aren't any functional changes in this PR at all, it's just whitespace fixes and minor reordering of lines based on the style (the `#include` lines for example).

### CODE REVIEW

**Comment by alistairking** [comment]
Thanks, this is much needed.
I have a slight preference for sticking with the clang format spec that's used in other CAIDA repos. E.g., https://github.com/CAIDA/libbgpstream/blob/master/.clang-format

**Comment by timwoj** [comment]
Oh! I looked at the top-level and didn't see one, so I went with LLVM since it was sorta close to what existed. I'll update the PR using that config.

**Review by alistairking — APPROVED** [review]
Thanks! This LGTM.

I'll let @digizeph or @kenkeys merge when they're satisfied.

**Comment by timwoj** [comment]
Ping on merging this. I'd like to rebase our version on this upstream one, but I want to wait until it's been officially merged.

**Comment by digizeph** [comment]
@kenkeys could you take a look at this PR?

**Comment by timwoj** [comment]
Any word on merging this?

**Comment by timwoj** [comment]
One more ping on this.

**Comment by timwoj** [comment]
Is there something still holding this up from getting merged?

**Pull request merged**
